### DISCUSSION TITLE
Question about cache with DNS load balancing

### DISCUSSION THREAD

**Discussion by MisteryMonster** [TechnitiumSoftware/DnsServer#450] [Q&A]
When I query using public DNS server with some domian name like `gb-courier-4.push-apple.com.akadns.net`, it will change the query result very frequently ( like changing every time when query ) even the TTL is not counted to zero, but after I set a forwarder in Technitium with the same public DNS server, query using Technitium will change very slow, even I set the TTL to a very low number like 5 seconds. The query log show the response type is all `Recursive` but did not changing the query result in about 1 minitue. Should it more alike I using the public DNS?

**Comment by ShreyasZare** [comment]
Thanks for asking. The stats and logs use the term `Recursive` for both recursion and forwarding. 

The thing you are seeing with response is quite normal. When you use Technitium DNS server, you get a cached response until the TTL expires. The response will only change when the cache is refreshed with new data.

While when you query to a public DNS sever, the response may change frequently since these public DNS providers have more than one DNS servers running behind a load balancer so each time you may get a response from a different server based on the contents of its cache.

Public DNS providers also use Anycast IP addresses. So it may happen that some times the request is getting routed to a different data center altogether.

**Reply by MisteryMonster** [reply]
I see, so when I get a response from Technitium DNS server forward from a public DNS server, let say this public DNS server reponse to Technitium DNS server a record have TTL=59s, then I go to `Settings→Cache` set the `Cache Maximum TTL` to 3s, it will still use the cached record in 59s because the recursive is still respect the TTL from public DNS server?

**Comment by ShreyasZare** [comment]
When you set Cache Maximum TTL value then it does not update entries that exists in the cache. Only new entries that are being added are updated with that max value. So you may want to clear cache to be sure of it.
